cli/tools/jupyter/mod.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+use crate::args::Flags;
+use crate::args::JupyterFlags;
+use crate::tools::repl;
+use crate::util::logger;
+use crate::CliFactory;
+use deno_core::anyhow::Context;
+use deno_core::error::AnyError;
+use deno_core::futures::channel::mpsc;
+use deno_core::op;
+use deno_core::resolve_url_or_path;
+use deno_core::serde::Deserialize;
+use deno_core::serde_json;
+use deno_core::Op;
+use deno_core::OpState;
+use deno_runtime::permissions::Permissions;
+use deno_runtime::permissions::PermissionsContainer;
+
+mod install;
+mod jupyter_msg;
+mod server;
+
+pub async fn kernel(
+  flags: Flags,
+  jupyter_flags: JupyterFlags,
+) -> Result<(), AnyError> {
+  let Some(connection_filepath) = jupyter_flags.conn_file else {
+    install::install()?;
+    return Ok(());
+  };
+
+  // This env var might be set by notebook
+  if std::env::var("DEBUG").is_ok() {
+    logger::init(Some(log::Level::Debug));
+  }
+
+  let factory = CliFactory::from_flags(flags).await?;
+  let cli_options = factory.cli_options();
+  let main_module =
+    resolve_url_or_path("./$deno$jupyter.ts", cli_options.initial_cwd())
+      .unwrap();
+  // TODO(bartlomieju): should we run with all permissions?
```

Comment:
I was kind of assuming that people could toggle which permissions are enabled via their kernelspec (or when they run the command to install the kernel).

Reply:
Well, we can certainly do that - however I'm wondering how user-friendly that would be. If you install kernelspec like `deno jupyter --install --allow-read --allow-write` and then hit a missing permission, you'd have to "reinstall" the kernel to elevate permissions like `deno jupyter --install --allow-read --allow-write --allow-env`. Maybe that's fine though?

Reply:
I think it's good to run with all permissions by default in a repl capacity like this. If someone wants to deploy a more locked down version they can change the flags (go the opposite direction on this).